internal/armada/server/event.go

```diff
@@ -49,13 +53,52 @@ func (s *EventServer) Report(ctx context.Context, message *api.EventMessage) (*t
 	if err := checkPermission(s.permissions, ctx, permissions.ExecuteJobs); err != nil {
 		return nil, status.Errorf(codes.PermissionDenied, "[Report] error: %s", err)
 	}
+	if event, ok := message.Events.(*api.EventMessage_Preempted); ok {
```

Comment:
Are we able to do this earlier in the process (i.e. as soon as the event comes in from the executor)? I ask because the new event ingester doesn't have access to the jobs db so would have difficult filling in here.